File: 5_자료구조(스택,큐,힙,해쉬 테이블)/9_1_아나그램(딕셔너리 해쉬)/answer2.js
const fs = require('fs')
const content = fs.readFileSync("in1.txt").toString();


let a_arr
let b_arr

content.trim().split('\n').map((v, i) => {

    i == 0 ? a_arr = v.trim().split('') : b_arr = v.trim().split('')

})


console.log(a_arr, b_arr)

let dic = {}

for (k of a_arr) {

    dic[k] = (dic[k] || 0) + 1
}

for (k of b_arr) {

    dic[k] = (dic[k] || 0) - 1
}


let result = "YES"

for (k of a_arr) {
    if (dic[k] > 0) {
        result = "NO"
        break
    }
}


console.log(result)

File: 5_자료구조(스택,큐,힙,해쉬 테이블)/9_1_아나그램(딕셔너리 해쉬)/answer2.py
import sys
sys.stdin = open("in4.txt", "r")
a = input()
b = input()

a_arr = list(map(str, str(a)))
b_arr = list(map(str, str(b)))

print(a_arr)
dic = dict()

for k in a_arr:
    dic[k] = dic.get(k, 0) + 1


for k in b_arr:
    dic[k] = dic.get(k, 0) - 1

# print(dic)

result = "YES"

for k in a_arr:
    if (dic.get(k) > 0):
        print("NO")
        break


print(result)
